blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -56,12 +118,69 @@ namespace blueprint
             _viewManager->performRootShadowTreeLayout();
         }
 
+        void paint(juce::Graphics& g) override
+        {
+            if (errorText)
+            {
+                g.fillAll(juce::Colour(0xffe14c37));
+                errorText->draw(g, getLocalBounds().toFloat().reduced(10.f));
+            }
+            else
+            {
+                View::paint(g);
+            }
+        }
+
         //==============================================================================
-        /** Evaluates a javascript bundle in the Ecmascript engine. */
+        /**
+         * Evaluates a javascript bundle/code in the Ecmascript engine.
+         *
+         * It is possible to pass a js bundle file to this evaluate() overload using juce::File::loadAsString. However,
+         * if you require hot-reload functionality you should use the juce::File based evaluate() overload.
+         * This overload can also be used to evaluate js code inline/directly.
+         **/
         juce::var evaluate (const juce::String& bundle)
         {
             JUCE_ASSERT_MESSAGE_THREAD
-            return engine.evaluate(bundle);
+
+            // Clear error state from previous js evals
+            errorText.reset();
+            auto result = engine.evaluate(bundle);
+
+            if (onBundleEvaluated)
+                onBundleEvaluated(std::nullopt);
+
+            return result;
+        }
+
+        /**
+         * Evaluates a javascript bundle file in the Ecmascript engine.
+         *
+         *  If hot-reload functionality has been enabled via enableHotReload(), ReactApplicationRoot will watch the
+         *  supplied bundle file for changes and provide instant UI updates. This is useful when working with Webpack
+         *  watchers etc.
+         *
+         *  Hot reloading is enabled by default in debug builds if JUCE_DEBUG is set.
+         **/
+        juce::var evaluate(const juce::File& bundle)
+        {
+            JUCE_ASSERT_MESSAGE_THREAD
+            jassert(bundle.existsAsFile());
+
+            if (hotReloadEnabled)
+            {
+                jassert(bundleWatcher);
+                bundleWatcher->watch(bundle);
+            }
+
+            auto result = evaluate(bundle.loadFileAsString());
+
+            //TODO: Should we guard against users calling evaluate on public engine member directly and missing
```

Comment:
@nick-thompson, this is a TODO that is probably worth you commenting on.

Reply:
Hm... good question. I think it's probably a "things you should know" section in the docs. I could imagine cases where you might want to evaluate something but don't care for any callbacks, especially in apps where almost all of the logic is done in js-land and callbacks to native aren't really important